modules/lang-painless/src/main/java/org/elasticsearch/painless/lookup/PainlessLookup.java

```diff
@@ -19,41 +19,129 @@
 
 package org.elasticsearch.painless.lookup;
 
-import java.util.Collection;
 import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
 
-/**
- * The entire API for Painless.  Also used as a whitelist for checking for legal
- * methods and fields during at both compile-time and runtime.
- */
-public final class PainlessLookup {
+import static org.elasticsearch.painless.lookup.PainlessLookupUtility.buildPainlessConstructorKey;
+import static org.elasticsearch.painless.lookup.PainlessLookupUtility.buildPainlessFieldKey;
+import static org.elasticsearch.painless.lookup.PainlessLookupUtility.buildPainlessMethodKey;
+import static org.elasticsearch.painless.lookup.PainlessLookupUtility.typeToCanonicalTypeName;
 
-    public Collection<Class<?>> getStructs() {
-        return classesToPainlessClasses.keySet();
-    }
+public final class PainlessLookup {
 
     private final Map<String, Class<?>> canonicalClassNamesToClasses;
     private final Map<Class<?>, PainlessClass> classesToPainlessClasses;
 
+    private final List<Class<?>> sortedClassesByCanonicalClassName;
```

Comment:
Why not have the canonicalClassNamesToClasses above be a TreeMap? Then their iteration order is sorted by the keys. But since this seems to only be used by the doc generator, I would keep the sorting to only occur in the doc generator, rather than making PainlessLookup always incur that cost.

Reply:
Agreed.  Fixed.